tree/20_cfe_basics/ncf_lib.cf

```diff
@@ -368,6 +368,58 @@ body classes classes_generic_return_codes_two(x, y, kept_return_code, repaired_r
                       "promise_kept_$(y)", "$(y)_kept", "$(y)_ok", "$(y)_not_repaired", "$(y)_reached" };
 }
 
+# Define x prefixes/suffixed with promises outcome, based on return codes
+body classes classes_generic_return_codes_two(x, y, kept_return_codes, repaired_return_codes)
```

Comment:
the name isn't really helpful in why it differs from the previous method
I forsee a lot of errors for users in the names of the methods (s or no s, that is the question ...)